python/14-pdf-numeric-table.ipynb

```diff
@@ -0,0 +1,314 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "# Detect Tables in raw text\n",
+    "\n",
+    "In this example using a threashold of 4 characters out of the 15 most used characters being digits works well to identify if the page is mostly constituted of numeric tables.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from collections import Counter\n",
+    "\n",
+    "\n",
+    "def containsatable(textcontent):\n",
+    "    counter = Counter(textcontent)\n",
+    "    mostcom = counter.most_common(15)\n",
+    "    countdigit=0\n",
+    "    for i in mostcom :\n",
+    "        if (i[0].isdigit()):\n",
+    "            countdigit=countdigit+1\n",
+    "    return(countdigit>=4)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from trp import Document\n",
+    "import math\n",
+    "doc = Document(response)\n",
+    "#print(doc.pages[0])\n",
+    "# Iterate over elements in the document\n",
+    "pagenum=0\n",
+    "pageswithtables=[]\n",
+    "for page in doc.pages:\n",
+    "    pagenum=pagenum+1\n",
+    "    text=''\n",
+    "    for line in page.lines:\n",
+    "        text=text+line.text\n",
+    "    if (containsatable(text)):\n",
+    "        pageswithtables.append(pagenum)\n",
+    "\n",
+    "print(\"List of pages with Tables:\")        \n",
+    "print (pageswithtables)\n",
+    "print(\"Percentage of pages with tables:\")    \n",
+    "print (str(len(pageswithtables))+\"/\"+str(pagenum))\n",
+    "print (str(math.trunc(len(pageswithtables)/pagenum*100))+\"%\")\n",
+    "    \n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Call Analyse Documents only on the pages with tables\n",
+    "First, we build a smaller PDF document containing only the pages identified as containing tables and then we run the AnalyzeDocument call on the new document."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from PyPDF2 import PdfReader, PdfWriter\n",
+    "\n",
+    "pdf_file_path = file\n",
+    "file_base_name = pdf_file_path.replace('.pdf', '')\n",
+    "\n",
+    "pdf = PdfReader(pdf_file_path)\n",
+    "\n",
+    "pages = pageswithtables\n",
+    "pdfWriter = PdfWriter()\n",
+    "\n",
+    "for page_num in pages:\n",
+    "    pdfWriter.add_page(pdf.pages[page_num-1])\n",
+    "\n",
+    "pdf_out = open(\"temp.pdf\", 'wb')\n",
+    "pdfWriter.write(pdf_out)\n",
+    "pdf_out.close()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!aws s3 cp temp.pdf s3://{data_bucket}/temp.pdf --only-show-errors"
+   ]
+  },
```

Comment:
Get it, this is why we need Sagemaker. Let's just add a comment that customers can use their own bucket for their own documents. For this sample I uploaded the temp.pdf to s3://amazon-textract-public-content/code-samples/DemoTableDocument-ExtractedTablePages.pdf, so we can reference that